### PR TITLE
Merge native synchronization package

### PR DESCRIPTION
**Pull request by devoncarew**
- move the `dart-lang/native_synchronization` repo and package into `pkgs/native_synchronization`
- done as part of https://github.com/dart-lang/labs/issues/2

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by natebosch — APPROVED** [review]

File: pkgs/native_synchronization/.gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Files and directories created by pub.
```
Comment (mosuem) [inline pkgs/native_synchronization/.gitignore:1]: Are some of these covered by the top-level `.gitignore`?

File: pkgs/native_synchronization/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: native_synchronization
+description: Low level synchronization primitives built on dart:ffi.
+version: 0.3.0
+repository: https://github.com/dart-lang/native_synchronization
```
Comment (mosuem) [inline pkgs/native_synchronization/pubspec.yaml:4]: Update to `labs/`

File: pkgs/native_synchronization/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name: native_synchronization
+description: Low level synchronization primitives built on dart:ffi.
+version: 0.3.0
```
Comment (mosuem) [inline pkgs/native_synchronization/pubspec.yaml:3]: Rev minor version for publish, pointing to the new repo.

**Review by mosuem — APPROVED** [review]
Needs an entry to the top-level readme table.

All these comments can be done in a follow up as well.

File: pkgs/native_synchronization/.gitignore (hunk 1 above)
Comment (devoncarew) [reply pkgs/native_synchronization/.gitignore:1]: We don't have a top-level gitignore here. And general, I think it makes sense for each package to be self-contained at least from the POV of having a complete gitignore file.

File: pkgs/native_synchronization/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+[![Dart](https://github.com/dart-lang/native_synchronization/actions/workflows/dart.yaml/badge.svg)](https://github.com/dart-lang/native_synchronization/actions/workflows/dart.yaml)
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply pkgs/native_synchronization/README.md:1]: 👍

File: pkgs/native_synchronization/pubspec.yaml (hunk 3 above)
Comment (devoncarew) [reply pkgs/native_synchronization/pubspec.yaml:3]: sgtm; will plan to do this as part of a follow-up PR

File: pkgs/native_synchronization/pubspec.yaml (hunk 2 above)
Comment (devoncarew) [reply pkgs/native_synchronization/pubspec.yaml:4]: sgtm; will plan to do this as part of a follow-up PR

**Pull request merged**